tensorflow_recommenders_addons/dynamic_embedding/python/keras/layers/embedding.py

```diff
@@ -0,0 +1,381 @@
+# Copyright 2021 The TensorFlow Recommenders-Addons Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# lint-as: python3
+"""
+Dynamic Embedding is designed for Large-scale Sparse Weights Training.
+See [Sparse Domain Isolation](https://github.com/tensorflow/community/pull/237)
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import tensorflow as tf
+
+from tensorflow.python.eager import context
+from tensorflow_recommenders_addons import dynamic_embedding as de
+from tensorflow_recommenders_addons.dynamic_embedding.python.ops import dynamic_embedding_variable as devar
+
+
+def _choose_reduce_method(combiner, sparse=False, segmented=False):
+  select = 'sparse' if sparse else 'math'
+  try:
+    module = getattr(tf, select)
+  except:
+    raise AttributeError('tensorflow has no attribute {}'.format(select))
+  select = 'segment_' if segmented else 'reduce_'
+  select += combiner
+  try:
+    method = getattr(module, select)
+  except:
+    raise AttributeError('Module [{}] has no attribute {}'.format(
+        module, select))
+  if not callable(method):
+    raise ValueError('{}: {} in {} is not callable'.format(
+        select, method, module))
+  return method
+
+
+@tf.function
+def reduce_pooling(x, combiner='sum'):
+  """
+  Default combine_fn for Embedding layer. By assuming input
+  ids shape is (batch_size, s1, ..., sn), it will get lookup result
+  with shape (batch_size, s1, ..., sn, embedding_size). Every
+  sample in a batch will be reduecd to a single vector, and thus
+  the output shape is (batch_size, embedding_size)
+  """
+
+  ndims = x.shape.ndims
+  combiner = _choose_reduce_method(combiner, sparse=False, segmented=False)
+
+  with tf.name_scope('deep_squash_pooling'):
+
+    if ndims == 1:
+      raise ValueError("reduce_pooling need at least dim-2 input.")
+    elif ndims == 2:
+      return combiner(x, 0)
+
+    for i in range(0, ndims - 2):
+      x = combiner(x, 1)
+    return x
+
+
+class Embedding(tf.keras.layers.Layer):
```

Comment:
Lack of `get_config` and `from_config` functions, making tf.keras.models.save_model function unavailable.
```
def get_config(self):
        config = {}
        base_config = super(EmbeddingLayer, self).get_config()
        return dict(list(base_config.items()) + list(config.items()))
```

Reply:
Accept